code/game/objects/structures/stool_bed_chair_nest/stools.dm

```diff
@@ -74,3 +73,15 @@
 		T.Weaken(5)
 		return
 	..()
+
+/obj/item/weapon/stool/dropped(mob/user) //Make sure we aren't leaving un-deconstructible stools.
+	..()
+	if(!user.in_throw_mode) //If this stool isn't getting thrown, turn it back into a structure.
+		origin.forceMove(user.loc)
+		qdel(src)
+
+/obj/item/weapon/stool/throw_impact(atom/hit_atom) //Thrown but uncaught stools will automatically turn into the structure version as soon as the throw is over.
+	..()
+	if(!iscarbon(loc)) //If it wasn't caught by someone, put the structure version down.
+		new origin.type(get_turf(src))

```

Comment:
Ah, okay, sounds like the problem is that `throw_impact` occurs twice if a dense object is hit - once for hitting the object, and again for hitting the floor. If this hits an object, `origin` would get moved out, this would get qdeleted (sending it to nullspace), then it would "impact" the floor, and move `origin` to nullspace. This code appears to fix the problem because you don't see the second stool getting created in nullspace.

Ensuring `origin` only ever gets moved once (null it out, check it for null before moving, maybe put all that into a helper proc to cut down on the copy-paste) should fix it. Sanity checking the target location (should be a turf), `origin`'s location (should be `src`), or for qdeletion would also fix it... but checking it for null will also make sure you don't runtime if `origin` was never actually set (e.g., admin spawned).